ballerina/error.bal

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2024 WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Represents any error related to Ballerina LDAP module
+public type Error distinct error<ErrorDetails>;
+
+# The error details type for the Ballerina LDAP module.
+#
+# + resultStatus - The status of the error
+# + message - The error message
+public type ErrorDetails record {|
+    string resultStatus?;
```

Comment:
```suggestion
    string resultCode?;
```
I think `result code` is the more familiar phrase in LDAP. WDYT?

Reply:
Fixed in [cdde708](https://github.com/ballerina-platform/module-ballerina-ldap/pull/45/commits/cdde70852f32464ef384105c5366885fa9486d98)